plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/register.ts

```diff
@@ -95,10 +95,13 @@ export function createCatalogRegisterAction(options: {
 
       ctx.logger.info(`Registering ${catalogInfoUrl} in the catalog`);
 
-      const result = await catalogClient.addLocation({
-        type: 'url',
-        target: catalogInfoUrl,
-      });
+      const result = await catalogClient.addLocation(
+        {
+          type: 'url',
+          target: catalogInfoUrl,
+        },
+        ...(ctx.token ? [{ token: ctx.token }] : []),
```

Comment:
```suggestion
        ctx.token ? { token: ctx.token } : {},
```

i see what the old code did, but unpacking as a vararg array so to speak felt less clear than this :)

Reply:
Just me, trying to learn the cool lingo the kids are using nowadays (and avoid changing the test request signature, but that ain't important). Changing.